src/python/pants/backend/terraform/target_gen.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from typing import Optional
+
+from pants.backend.terraform.target_types import (
+    TerraformModule,
+    TerraformModules,
+    TerraformModulesSources,
+)
+from pants.build_graph.address import Address
+from pants.core.goals.tailor import group_by_dir
+from pants.engine.internals.selectors import Get
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import (
+    GeneratedTargets,
+    GenerateTargetsRequest,
+    HydratedSources,
+    HydrateSourcesRequest,
+    ImmutableValue,
+    Sources,
+    Target,
+)
+from pants.engine.unions import UnionRule
+
+
+class GenerateTerraformModuleTargetsRequest(GenerateTargetsRequest):
+    generate_from = TerraformModules
+
+
+@rule
+async def generate_terraform_module_targets(
+    request: GenerateTerraformModuleTargetsRequest,
+) -> GeneratedTargets:
+    # Hydrate the sources referenced by the generator.
+    sources = await Get(
+        HydratedSources, HydrateSourcesRequest(request.generator.get(TerraformModulesSources))
+    )
```

Comment:
`SourcesPaths` is better so you don't digest the files. All you need is file paths.

Reply:
Done.